pkg/daemon/certs/certs.go

```diff
@@ -0,0 +1,125 @@
+package certs
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"syscall"
+	"unsafe"
+
+	syswindows "golang.org/x/sys/windows"
+	"k8s.io/klog/v2"
+)
+
+// Reconcile ensures the certificates in the given trusted CA bundle file are imported as system certificates.
+// Returns a boolean whether any certs have needed to be imported
+func Reconcile(trustBundle string) (bool, error) {
+	if trustBundle == "" {
+		// TODO: ensure that all certs that we've added previously are removed https://issues.redhat.com/browse/WINC-998
+		return false, nil
+	}
+	// Read expected certs from CA trust bundle file
+	pemData, err := ioutil.ReadFile(trustBundle)
+	if err != nil {
+		klog.Errorf("Failed to read certificate file: %v", err)
+	}
+	// Split the file by certificate blocks
+	certBlocks := splitPEMBlocks(pemData)
+
+	// Open the root certificate store
+	store, err := syswindows.CertOpenStore(syswindows.CERT_STORE_PROV_SYSTEM, 0, 0,
+		syswindows.CERT_SYSTEM_STORE_LOCAL_MACHINE, uintptr(unsafe.Pointer(syscall.StringToUTF16Ptr("ROOT"))))
+	if err != nil {
+		return false, fmt.Errorf("Failed to open root certificate store: %v", err)
+	}
+	defer syswindows.CertCloseStore(store, 0)
+
+	// Get all existing from system store
+	existingSystemCerts, err := getAllCerts(store)
+	if err != nil {
+		klog.Errorf("Failed to read certificate file: %v", err)
+	}
+
+	certChange := false
+	for _, certBlock := range certBlocks {
+		// Decode and parse the certificate
+		cert, err := x509.ParseCertificate(certBlock.Bytes)
+		if err != nil {
+			klog.Errorf("Failed to parse certificate %s: %v\n", cert.Subject.String(), err)
+			continue
+		}
+		if !containsCert(existingSystemCerts, cert) {
+			// Convert x509 certificate to a Windows CertContent
+			certContext, err := syswindows.CertCreateCertificateContext(
+				syswindows.X509_ASN_ENCODING|syswindows.PKCS_7_ASN_ENCODING, &cert.Raw[0], uint32(len(cert.Raw)))
+			if err != nil {
+				klog.Errorf("Failed to create content from x509 cert %s: %v\n", cert.Subject.String(), err)
+				continue
+			}
+			// Add the certificate to the instances's system-wide trust store
+			if err = syswindows.CertAddCertificateContextToStore(store, certContext,
+				syswindows.CERT_STORE_ADD_REPLACE_EXISTING, nil); err != nil {
+				klog.Errorf("Failed to import certificate %s: %v\n", cert.Subject.String(), err)
+				continue
+			}
+			certChange = true
+			klog.Infof("Certificate %s imported successfully.", cert.Subject.String())
+		}
```

Comment:
will we retry the cert import if we don't return errors?

Reply:
yes, it'll retry any failed ones on the next WICD controller Reconcile() call (every 30 seconds)